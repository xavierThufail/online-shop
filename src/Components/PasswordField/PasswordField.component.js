import React from 'react';
import { Input, FormItem } from 'formik-antd';
import { useField } from 'formik';

import config from './PasswordField.config';
import './PasswordField.component.css';

const handleOnChange = setValue => e => setValue(e.target.value);

const TextField = ({ formStyle, inputStyle, placeholder, label, maxLength, ...props }) => {
  const [field, meta, { setValue }] = useField(props);
  const isError = meta.touched && meta.error;

  return (
    <FormItem
      validateStatus={isError ? 'error' : ''}
      help={isError ? meta.error : ''}
      label={label}
      colon={false}
      style={formStyle}
    >
      <Input.Password
        style={inputStyle}
        placeholder={placeholder}
        maxLength={maxLength}
        {...field}
        onChange={handleOnChange(setValue)}
        visibilityToggle={false}
      />
    </FormItem>
  );
};

TextField.propTypes = config.propTypes;
TextField.defaultProps = config.defaultProps;
TextField.displayName = config.displayName;

export default TextField;
