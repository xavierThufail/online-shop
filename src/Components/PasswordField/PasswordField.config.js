import PropTypes from 'prop-types';

const propTypes = {
  inputStyle: PropTypes.object,
  formStyle: PropTypes.object,
  name: PropTypes.string.isRequired,
  label: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  placeholder: PropTypes.string,
  maxLength: PropTypes.number
};

const defaultProps = {
  label: '',
  style: {},
  placeholder: ''
};

const displayName = 'TextField';

const config = {
  propTypes,
  defaultProps,
  displayName
};

export default config;
