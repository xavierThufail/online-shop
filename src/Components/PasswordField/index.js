export { default as PasswordField } from './PasswordField.component';
