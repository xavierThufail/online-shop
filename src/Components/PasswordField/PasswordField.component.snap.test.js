import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, act, fireEvent } from '@testing-library/react';
import { Formik } from 'formik';

import PasswordField from './PasswordField.component';
import '../../matchMedia.mock';

describe('PasswordField component snap test', () => {

  beforeEach(jest.clearAllMocks);

  it('Render PasswordField, should matching snapshot', () => {
    const mockedProps = {
      name: 'test'
    };
    
    const { container } = render(
      <Formik>
        {formikProps => <PasswordField {...formikProps} {...mockedProps} />}
      </Formik>
    );

    expect(container).toMatchSnapshot();
  });

  it('Render PasswordField with label, should matching snapshot', () => {
    const mockedProps = {
      label: 'Username',
      name: 'username',
      placeholder: 'username'
    };

    const { container, getByText, getByPlaceholderText } = render(
      <Formik>
        {formikProps => <PasswordField {...formikProps} {...mockedProps} />}
      </Formik>  
    );
    const label = getByText(mockedProps.label);
    const input = getByPlaceholderText(mockedProps.placeholder);

    act(() => {
      fireEvent.blur(input);
    });

    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
