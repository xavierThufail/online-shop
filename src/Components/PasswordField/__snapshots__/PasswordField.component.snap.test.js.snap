// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordField component snap test Render PasswordField with label, should matching snapshot 1`] = `
<div>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col ant-form-item-label"
    >
      <label
        class="ant-form-item-no-colon"
        title="Username"
      >
        Username
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <input
            action="click"
            class="ant-input ant-input-password"
            name="username"
            placeholder="username"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-form-item-explain"
      >
        <div
          role="alert"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`PasswordField component snap test Render PasswordField, should matching snapshot 1`] = `
<div>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <input
            action="click"
            class="ant-input ant-input-password"
            name="test"
            placeholder=""
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-form-item-explain"
      >
        <div
          role="alert"
        />
      </div>
    </div>
  </div>
</div>
`;
